openeogeotrellis/logs.py

```diff
@@ -1,67 +1,136 @@
                                  message=f"The value passed for the query parameter 'offset' is invalid: {offset}")
 
 
-def _elasticsearch_logs(job_id: str, search_after: Optional[list]) -> Iterable[dict]:
-    page_size = 100
+def _elasticsearch_logs(
+    job_id: str,
+    create_time: Optional[dt.datetime] = None,
+    search_after: Optional[list] = None,
+) -> Iterable[dict]:
+    """Internal helper function to retrieve a job's logs from Elasticsearch.
 
-    with Elasticsearch(ES_HOSTS) as es:
-        while True:
-            search_result = es.search(
-                index=ES_INDEX_PATTERN,
-                query={
-                    'bool': {
-                        'filter': [{
-                            'term': {
-                                'job_id': job_id
-                            }
-                        }, {
-                            'terms': {
-                                'tags': ES_TAGS
-                            }
-                        }]
-                    }
-                },
-                search_after=search_after,
-                size=page_size,
-                sort=[
-                    {'@timestamp': {'order': 'asc'}},
-                    {'log.offset': {'order': 'asc'}},  # tie-breaker
-                    # chances are slim that two processes write on the same line within the same millisecond
-                    # 'log.file.path', 'host.name'
-                ],
-                request_timeout=60
-            )
+    :param job_id:
+        ID of the Job
+
+    :param create_time:
+        Time the job was created, only log records starting from that time will be retrieved
 
-            hits = search_result['hits']['hits']
+    :param search_after:
+        Search only after this offset.
+        This offset is a list which takes two elements: the timestamp and log.offset
+        where timestamp is the Unix Epoch (int) and log.offset (int) in Elasticsearch.
 
-            for hit in hits:
-                search_after = hit['sort']
+        For example: [1673351608383, 102790]
 
-                # Skip the log line if the log level is empty
-                entry = _as_log_entry(log_id=json.dumps(search_after), hit=hit)
-                if "level" in entry:
-                    yield entry
+    :raises OpenEOApiException:
+        - Either when the offset is not valid JSON
+        - or when Elasticsearch had a connection timeout
 
-            if len(hits) < page_size:
-                break
+    :return: an generator that yields a dict for each log record.
+    """
+
+    page_size = 100
+    query = {
+        "bool": {
+            "filter": [
+                {"term": {"job_id": job_id}},
+                {"terms": {"tags": ES_TAGS}},
+            ],
+        }
+    }
+    if create_time:
+        query["bool"]["filter"].append(
+            {
+                "range": {
+                    "@timestamp": {
+                        "format": "strict_date_optional_time",
```

Comment:
Should be "strict_date_time".

Reply:
Thanks for catching that one. I have mistakenly re-introduced the wrong format.

Reply:
Done.